src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -2,13 +2,78 @@
 
 import core.basesyntax.dao.StorageDao;
 import core.basesyntax.dao.StorageDaoImpl;
+import core.basesyntax.exceptions.RegistrationServiceException;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private static final int MIN_LENGTH = 6;
+    private static final int MIN_AGE = 18;
+
     private final StorageDao storageDao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
-        return null;
+        checkLoginNotNull(user);
+        checkLoginMinLength(user);
+        checkPasswordNotNull(user);
+        checkPassword(user);
+        checkAgeNotNull(user);
+        checkAge(user);
```

Comment:
not critical, but all of that could be in a single `validateUser()` method, to not call them one by one here